src/arcade_collection/output/parse_growth_file.py

```diff
@@ -0,0 +1,143 @@
+import json
+from os import path
+import tarfile
+from typing import List, Union
+
+import ntpath
+import numpy as np
+import pandas as pd
+from prefect import task
+
+
+GROWTH_COLUMNS = [
+    "TICK",
+    "SEED",
+    "U",
+    "V",
+    "W",
+    "Z",
+    "POSITION",
+    "POPULATION",
+    "STATE",
+    "VOLUME",
+    "CYCLE",
+]
+
+
+@task
+def parse_growth_file(tar: tarfile.TarFile) -> pd.DataFrame:
+    """
+    Parse the tumor growth tar file.
+
+    Parameters
+    ----------
+    tar :
+        Tar file of simulations.
+
+    Returns
+    -------
+    :
+        Data of all timepoints of all simulations in tar file.
+    """
+    all_timepoints = []
+    for member in tar.getmembers():
+        extracted_member = tar.extractfile(member)
+        if extracted_member is not None:
+            base = ntpath.basename(member.name)
+            base_file = path.splitext(base)
+            file_name = base_file[0]
+            extension = base_file[1]
+            if file_name[0] != "." and extension == ".json":
+                extracted_json = json.loads(extracted_member.read().decode("utf-8"))
+                seed = extracted_json["seed"]
+
+                for timepoint in extracted_json["timepoints"]:
+                    one_timepoint = parse_growth_timepoint(timepoint, seed)
+                    for data in one_timepoint:
+                        all_timepoints.append(data)
+
+    timepoints_df = pd.DataFrame(all_timepoints, columns=GROWTH_COLUMNS)
+    return timepoints_df
+
+
+def convert_state_to_string(state_index: int) -> Union[str, None]:
+    """
+    Convert the numbers that represent cell state into an annotation.
+
+    Parameters
+    ----------
+    state_index :
+        The index of cell states.
+
+    Returns
+    -------
+    :
+        The cell state annotation.
+    """
+    if state_index == 0:
+        return "NEU"
+    elif state_index == 1:
+        return "APO"
+    elif state_index == 2:
+        return "QUI"
+    elif state_index == 3:
+        return "MIG"
+    elif state_index == 4:
+        return "PRO"
+    elif state_index == 5:
+        return "SEN"
+    elif state_index == 6:
+        return "NEC"
+
+    return None
+
+
+def parse_growth_timepoint(timepoint: dict, seed: int) -> List[list]:
+    """
+    Parse one timepoint of the simulation
```

Comment:
Could be more descriptive in docstrings: what was the original data structure and what is the new data structure? What modifications are being made (e.g. averaging, rounding) to the data?